### PR TITLE
Move C++ SSL transport to Ice core

### PR DESCRIPTION
**Pull request by pepone**

### CODE REVIEW

File: cpp/src/IceSSL/RFC2253.h [hunk 1]
```diff
@@ -22,41 +23,38 @@
 // results in the pair ("O","Sue\, Grabit and Runn") and not
 // ("O","Sue, Grabit and Runn").
 //
-namespace IceSSL
+namespace IceSSL::RFC2253
 {
-    namespace RFC2253
+    typedef std::list<std::pair<std::string, std::string>> RDNSeq;
```
Comment (externl) [inline cpp/src/IceSSL/RFC2253.h:28]:
```suggestion
    using RDNSeq = std::list<std::pair<std::string, std::string>>;
```

File: cpp/src/IceSSL/RFC2253.h [hunk 2]
```diff
@@ -22,41 +23,38 @@
...
-        //
-        // Unescape the string.
-        //
-        ICESSL_API std::string unescape(const std::string&);
-
-    }
+        RDNSeq rdn;
+        bool negate;
+    };
+    typedef std::list<RDNEntry> RDNEntrySeq;
```
Comment (externl) [inline cpp/src/IceSSL/RFC2253.h:35]:
```suggestion
    using RDNEntrySeq = std::list<RDNEntry>;
```

**Review by externl — APPROVED** [review]
Looks good to me!

**Comment by pepone** [comment]
I would fix review-comments in a follow-up PR.

**Pull request merged**